src/OpenLoco/GameCommands.cpp

```diff
@@ -120,89 +34,99 @@ namespace OpenLoco::GameCommands
-        nullptr,               // 49                                 0x00496C22
-        nullptr,               // 50                                 0x0049711F
-        nullptr,               // 51                                 0x004A6FDC
-        nullptr,               // 52                                 0x004A734F
-        nullptr,               // 53                                 0x0047AF0B
-        nullptr,               // remove_industry = 54,              0x0042ECFC
-        nullptr,               // build_company_headquarters = 55,   0x0042EEAF
-        nullptr,               // 56                                 0x00492C41
-        nullptr,               // 57                                 0x00493559
-        nullptr,               // 58                                 0x004267BE
-        nullptr,               // vehicle_abort_pickup_air = 59,     0x00426B29
-        nullptr,               // 60                                 0x00493AA7
-        nullptr,               // 61                                 0x00494570
-        nullptr,               // 62                                 0x0042773C
-        nullptr,               // vehicle_abort_pickup_water = 63,   0x004279CC
-        nullptr,               // 63                                 0x0042F6DB
-        nullptr,               // 64                                 0x00435506
-        nullptr,               // change_company_face = 66,          0x00469CCB
-        nullptr,               // load_multiplayer_map = 67,         0x00444DA0
-        nullptr,               // 68                                 0x0046F8A5
-        nullptr,               // 69                                 0x004454BE
-        nullptr,               // 70                                 0x004456C8
-        nullptr,               // send_chat_message = 71,            0x0046F976
-        nullptr,               // multiplayer_save = 72,             0x004A0ACD
-        nullptr,               // update_owner_status = 73,          0x004383CA
-        nullptr,               // vehicle_speed_control = 74,        0x004BAB63
-        nullptr,               // vehicle_order_up = 75,             0x00470CD2
-        nullptr,               // vehicle_order_down = 76,           0x00470E06
-        nullptr,               // vehicle_apply_shunt_cheat = 77,    0x004BAC53
-        nullptr,               // apply_free_cash_cheat = 78,        0x00438A08
-        nullptr,               // rename_industry = 79,              0x00455029
-        Vehicles::cloneVehicle // vehicle_clone = 80,                *NEW*
+
+    struct GameCommandInfo
+    {
+        GameCommandFunc implementation;
+        uintptr_t originalAddress; // original array: 0x004F9548
+        bool canUseWhilePaused;    // original array: 0x004F9688
+    };
+
+    // clang-format off
+    static constexpr GameCommandInfo _gameCommandDefinitions[81] = {
+        { nullptr,                0x004AF1DF, true  }, // vehicle_rearrange = 0,
```

Comment:
Why not make it a map and use an enum as key? That way the 'comments' are always correct.

Reply:
I would try to avoid the map, linear search on this is much faster, not that performance matters much here but I also think that it reads a bit nicer.

Here is how I would do it:
```
{ Id, Function, Address, { Attributes } }
```

Reply:
The identifier would only be there for human readability and maintainability, not actually for looking up, right? At least, as it stands currently.

So, concretely, you're suggesting something like the snippet below?
```cpp
{ GameCommand::vehicle_pickup, nullptr, 0x004B0826, true  },
```
We'd have to add placeholders to the enum as well; a lot of the other commands are still missing.

Reply:
@AaronVanGeffen As long as it's in compiled code, I'm fine with it. We could add some sort of "validate that key matches index" function at the start of the game (during compiling maybe even?)